### ISSUE TITLE
Help with starting demo

### ISSUE THREAD

**Issue by johnsonjo4531** [benjipott/videogular-chromecast#1]
Hey looks like a great plugin/module that you have. I was trying to run the demo, and implement it into my own app using videogular.  (I've honestly been working on it for the past 2-3 hours, and a couple of hours on past days. I thought I would finally ask.) I was wondering if you could help me I am new to grunt and to bower and am not sure how they work together in the scope of your project. My main question is how to start the demo page you have.

  I have a cloned repository and all I want to do is see the video player run in the browser with the chromecast part working.

What I've tried:
  I have tried running bower install from the the root of this repository that I cloned on my pc. bower install did nothing. So I decided to bower install all the required videogular dependencies that are on the readme myself using bower install package-name. I tried running grunt and grunt connect as well as grunt wiredep with no luck on figuring out how to change the index.html into a format so that I could use it or get a server up and running that would serve the file in it's usable format.  I'm probably doing a lot of things wrong and any help would be appreciated.  It would be nice if there was a readme for starting the demo. 

Any help would be greatly appreciated.

**Comment by benjipott** [comment]
ok I'll make you a getstarted section in the README as possible

**Comment by benjipott** [comment]
1. Sign in to the Google Cast SDK Developer Console and register your Cast application
   https://developers.google.com/cast/docs/registration
2. Add cast sdk in head of your html main file

``` html
<script type="text/javascript" src="https://www.gstatic.com/cv/js/sender/v1/cast_sender.js"></script>
```
1. add your cast id

``` html
 <vg-media vg-chromecast="yourCastId">
 </vg-media>
```

**Comment by johnsonjo4531** [comment]
Thanks I appreciate it. I'll try that. For now I think I'll just use the Default Media Receiver which I noticed you fall back to if the castId isn't provided.  I still am having a little trouble getting everything up and running.  I've gotten the video player to atleast start working and the cast button to show up I can press the button, but when I click to start casting it brings up the cast receiver on my tv, but doesn't start the video. It may be because I'm playing the video from the demo. I'm not sure if that's the problem, but I'll follow your readme and see if that will help.

**Comment by benjipott** [comment]
Do you have an error in the console?

**Comment by johnsonjo4531** [comment]
Uncaught DataCloneError: Failed to execute 'postMessage' on 'Window': An object could not be cloned.

```
chrome.cast.dc.wf   @   cast_sender.js:5919
b.Ka    @   cast_sender.js:6162
b.Hi    @   cast_sender.js:6121
b.Ji    @   cast_sender.js:6130
b.Mp    @   cast_sender.js:6135
chrome.cast.q.Co    @   cast_sender.js:6324
link.scope.onSessionSuccess @   videogular-chromecast.js:162
b.Qo    @   cast_sender.js:5996
b.mp    @   cast_sender.js:6017
chrome.cast.dc.bp   @   cast_sender.js:5916
```

This is my loadRequest variable value JSON.stringified

``` javascript
"{
    "type":"LOAD",
    "requestId":null,
    "sessionId":"C4438D8C-9166-4C45-96A4-F5CA5327EDF4",
    "media":{
        "contentId":{},
        "streamType": "BUFFERED",
        "contentType":"video/mp4",
        "metadata":{
            "type": 0,
            "metadataType": 0,
            "title":"test",
            "subtitle":null,
            "images":null,
            "releaseYear":null,
            "releaseDate":null,
            "description":"a simple video cast"
        },
        "duration":null,
        "tracks":null,
        "textTrackStyle":null,
        "customData":null
    },
    "activeTrackIds":null,
    "autoplay":true,
    "currentTime":0,"
    customData":null
}"
```

**Comment by benjipott** [comment]
do you use a local server on // localhost ? this error occurs if you start the page file: //

**Comment by johnsonjo4531** [comment]
Yes, I'm using a local node.js server and I'm accessing the page with videogular-chromecast through localhost.

I'm using the default videogular code just until I can get it working. Hotlinking the video wouldn't cause a problem would it?

``` javascript
.controller('HomeCtrl',
    ["$sce", function ($sce) {
        this.config = {
            preload: "none",
            sources: [
                {src: $sce.trustAsResourceUrl("http://static.videogular.com/assets/videos/videogular.mp4"), type: "video/mp4"},
                {src: $sce.trustAsResourceUrl("http://static.videogular.com/assets/videos/videogular.webm"), type: "video/webm"},
                {src: $sce.trustAsResourceUrl("http://static.videogular.com/assets/videos/videogular.ogg"), type: "video/ogg"}
            ],
            tracks: [
                {
                    src: "http://www.videogular.com/assets/subs/pale-blue-dot.vtt",
                    kind: "subtitles",
                    srclang: "en",
                    label: "English",
                    default: ""
                }
            ],
            theme: {
                url: "http://www.videogular.com/styles/themes/default/latest/videogular.css"
            },
            plugins: {
                poster: "http://www.videogular.com/assets/images/videogular.png"
            }
        };
    }]
)

```

I'll look into it more when I get home and I'll step through the google code and try and figure it out with a little research and if that doesn't work out then I'll at least give you better feedback from why my code isn't working. Hey thanks for your help with this issue so far.

**Comment by simsketch** [comment]
I am also receiving this error message:

Uncaught DataCloneError: Failed to execute 'postMessage' on 'Window': An object could not be cloned.

**Comment by simsketch** [comment]
OK, I resolved my issue by putting

<script src="lib/bower-videogular-chromecast/dist/videogular-chromecast.js"></script>

above this line

<script type="text/javascript" src="https://www.gstatic.com/cv/js/sender/v1/cast_sender.js"></script>

**Comment by simsketch** [comment]
Actually, I am still having this issue. I am going to try invoke chromecast separately to see if I get the same error...I will report back soon.

**Comment by simsketch** [comment]
Got a new wireless router, that didn't change anything.

**Comment by simsketch** [comment]
still experiencing the same error, but i will try updating to the latest repo or reimplementing.
